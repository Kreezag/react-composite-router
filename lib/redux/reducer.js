// ACTIONS
const CHANGE_STATE = '@@router/CHANGE_STATE';



// ACTION CREATORS
export const changeStateAction = (name, params) => ({
  type: CHANGE_STATE,
  payload: { name, params }
});



// REDUCER
export default (state = { name: '', params: {} }, action) => {
  if (action.type === CHANGE_STATE) {
    return Object.assign({}, state, {
      name: action.payload.name,
      params: action.payload.params
    });
  }

  return state;
};
