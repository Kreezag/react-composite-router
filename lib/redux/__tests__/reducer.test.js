/* eslint-env browser, jest */

import reducer, { changeStateAction } from '../reducer';

describe('redux reducer', () => {
  it('should set default values', () => {
    const state = reducer(undefined, { type: 'some' });

    expect(state).toMatchSnapshot();
  });

  it('should update state', () => {
    const state = reducer(undefined, changeStateAction('some', { param: 123 }));

    expect(state).toMatchSnapshot();
  });
});
