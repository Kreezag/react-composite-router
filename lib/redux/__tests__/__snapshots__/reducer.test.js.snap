exports[`redux reducer should set default values 1`] = `
Object {
  "name": "",
  "params": Object {},
}
`;

exports[`redux reducer should update state 1`] = `
Object {
  "name": "some",
  "params": Object {
    "param": 123,
  },
}
`;
